src/Views/MainView.vala

```diff
@@ -77,10 +96,43 @@ public class Sideload.MainView : AbstractView {
         });
     }
 
-    public void display_details (string? size) {
+    public void display_details (string? size, bool extra_repo) {
         if (size != null) {
-            download_size_label.label = _("Download size up to: %s").printf (size);
-            details_stack.visible_child_name = "details";
+            download_size_label.label = _("Download size may be up to %s").printf (size);
+        } else {
+            download_size_label.label = _("Unknown download size");
         }
+
+        if (extra_repo) {
+            var updates_icon = new Gtk.Image.from_icon_name ("system-software-update-symbolic", Gtk.IconSize.BUTTON);
```

Comment:
I think this should actually always be displayed. Even if the repo is already added, any updates to this app from existing repos will still not be reviewed.